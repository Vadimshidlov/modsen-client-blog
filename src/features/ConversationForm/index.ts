export { ConversationForm } from '@/features/ConversationForm/ui/ConversationForm';
