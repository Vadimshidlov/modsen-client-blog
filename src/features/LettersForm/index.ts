export { LettersForm } from '@/features/LettersForm/ui/LettersForm';
