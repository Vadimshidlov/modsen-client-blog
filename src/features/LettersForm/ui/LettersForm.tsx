'use client';

import React, { useEffect, useRef } from 'react';
import { ErrorMessage, Field, Form, Formik, FormikHelpers, FormikValues } from 'formik';
import emailjs from '@emailjs/browser';
import styles from '@/features/LettersForm/ui/LettersForm.module.scss';
import { Button } from '@/shared/ui/Button';
import { lettersFormScheme } from '@/features/LettersForm/constants/scheme';
import { toast } from 'react-toastify';
import { useTranslations } from 'next-intl';

export type LettersFormType = {
  email: string;
};

export function LettersForm() {
  const t = useTranslations('LettersForm');
  const buttonText = t('buttonText');

  const formRef = useRef<HTMLFormElement>(null);

  useEffect(() => emailjs.init(process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY as string), []);

  const sendEmail = (
    data: FormikValues,
    { setSubmitting, resetForm }: FormikHelpers<LettersFormType>,
  ) => {
    setSubmitting(true);

    if (!formRef.current || data.email === '') return;

    emailjs
      .send(
        process.env.NEXT_PUBLIC_EMAILJS_SERVICE_ID as string,
        process.env.NEXT_PUBLIC_EMAILJS_TEMPLATE_TWO as string,
        {
          name: 'Vadim Shydlou',
          email: data.email,
        },
      )
      .then(
        () => {
          toast.success('Subscription is successful!');
          resetForm();
        },
        () => {
          toast.error('Error with subscription!');
        },
      )
      .finally(() => {
        setSubmitting(false);
      });
  };

  return (
    <div className={styles.form__container}>
      <h3 className={styles.form__title}>{t('title')}</h3>
      <Formik
        validationSchema={lettersFormScheme}
        initialValues={{ email: '' }}
        onSubmit={sendEmail}
      >
        {({ isSubmitting }) => (
          <Form className={styles.form} ref={formRef}>
            <label htmlFor="email" className={styles.formLabel}>
              <Field
                type="text"
                name="email"
                id="email"
                className={styles.formInput}
                placeholder={t('inputPlaceholder')}
                autoComplete="off"
              />
              <ErrorMessage className={styles.validationError} name="email" component="div" />
            </label>
            <Button isFormButton isSubmitting={isSubmitting} disabled={isSubmitting}>
              {isSubmitting ? 'Loading...' : `${buttonText}`}
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  );
}
