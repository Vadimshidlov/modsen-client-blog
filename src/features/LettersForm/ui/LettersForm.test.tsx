import React from 'react';
import { render } from '@testing-library/react';
import { LettersForm } from '@/features/LettersForm';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('LettersForm component', () => {
  test('should matches snapshot', () => {
    const { container } = render(<LettersForm />);
    expect(container).toMatchSnapshot();
  });
});
