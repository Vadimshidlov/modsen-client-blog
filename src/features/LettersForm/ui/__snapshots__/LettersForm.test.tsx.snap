// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LettersForm component should matches snapshot 1`] = `
<div>
  <div
    class="form__container"
  >
    <h3
      class="form__title"
    >
      Mocked title
    </h3>
    <form
      action="#"
      class="form"
    >
      <label
        class="formLabel"
        for="email"
      >
        <input
          autocomplete="off"
          class="formInput"
          data-testid="letters-form-imput"
          id="email"
          name="email"
          placeholder="Mocked inputPlaceholder"
          type="text"
          value=""
        />
      </label>
      <button
        class="button fullWidth"
        type="submit"
      >
        Mocked buttonText
      </button>
    </form>
  </div>
</div>
`;
